service_capacity_modeling/models/org/netflix/elasticsearch.py

```diff
@@ -0,0 +1,471 @@
+    if cluster.cluster_params:
+        cluster.cluster_params.update(params)
+    else:
+        cluster.cluster_params = params
+
+
+# pylint: disable=too-many-locals
+def _estimate_elasticsearch_cluster_zonal(
+    instance: Instance,
+    drive: Drive,
+    desires: CapacityDesires,
+    zones_per_region: int = 3,
+    copies_per_region: int = 3,
+    max_local_disk_gib: int = 4096,
+    max_regional_size: int = 240,
+    max_rps_to_disk: int = 500,
+) -> Optional[CapacityPlan]:
+
+    # Netflix Elasticsearch doesn't like to deploy on really small instances
+    if instance.cpu < 2 or instance.ram_gib < 14:
+        return None
+
+    # (FIXME): Need elasticsearch input
+    # Right now Elasticsearch doesn't deploy to cloud drives, just adding this
+    # here and leaving the capability to handle cloud drives for the future
+    if instance.drive is None:
+        return None
+
+    rps = desires.query_pattern.estimated_read_per_second.mid // zones_per_region
+
+    # Based on the disk latency and the read latency SLOs we adjust our
+    # working set to keep more or less data in RAM. Faster drives need
+    # less fronting RAM.
+    ws_drive = instance.drive or drive
+    working_set = working_set_from_drive_and_slo(
+        drive_read_latency_dist=dist_for_interval(ws_drive.read_io_latency_ms),
+        read_slo_latency_dist=dist_for_interval(
+            desires.query_pattern.read_latency_slo_ms
+        ),
+        estimated_working_set=desires.data_shape.estimated_working_set_percent,
+        # Elasticsearch has looser latency SLOs, target the 90th percentile of disk
+        # latency to keep in RAM.
+        target_percentile=0.90,
+    ).mid
+
+    requirement = _estimate_elasticsearch_requirement(
+        instance=instance,
+        desires=desires,
+        working_set=working_set,
+        reads_per_second=rps,
+        zones_per_region=zones_per_region,
+        copies_per_region=copies_per_region,
+        max_rps_to_disk=max_rps_to_disk,
+    )
+
+    base_mem = (
+        desires.data_shape.reserved_instance_app_mem_gib
+        + desires.data_shape.reserved_instance_system_mem_gib
+    )
+
+    cluster = compute_stateful_zone(
+        instance=instance,
+        drive=drive,
+        needed_cores=int(requirement.cpu_cores.mid),
+        needed_disk_gib=int(requirement.disk_gib.mid),
+        needed_memory_gib=int(requirement.mem_gib.mid),
+        needed_network_mbps=requirement.network_mbps.mid,
+        # Assume that by provisioning enough memory we'll get
+        # a 90% hit rate, but take into account the reads per read
+        # from the per node dataset using leveled compaction
+        # FIXME: I feel like this can be improved
+        required_disk_ios=lambda x: _es_io_per_read(x) * math.ceil(0.1 * rps),
+        # Elasticsearch requires ephemeral disks to be % full because leveled
+        # compaction can make progress as long as there is some headroom
+        required_disk_space=lambda x: x * 1.25,
```

Comment:
Existing ES self-service provisions clusters so that data instances have at least 30% of free space left.

Reply:
Ok changing to 1.4 (1/ 0.7)